File: benchmarks/python/schwefel222fcn.py
import numpy as np

def schwefel222fcn(x: np.ndarray) -> np.ndarray:
    """
    Computes the value of the Schwefel 2.22 function.
    
    Parameters:
    -----------
    x : numpy.ndarray
        Input array of size M-by-N.
        
    Returns:
    --------
    scores : numpy.ndarray
        Vector of size M-by-1 in which each row contains the function value
        for the corresponding row of x.
        
    Author: Mazhar Ansari Ardeh
    Please forward any comments or bug reports to mazhar.ansari.ardeh at
    Google's e-mail service or feel free to kindly modify the repository.
    """
    absx = np.abs(x)
    scores = np.sum(absx, axis=1) + np.prod(absx, axis=1)
    return scores

File: src/matlab/schwefel222fcn.m
% Computes the value of the Schwefel 2.22 function.
% SCORES = SCHWEFEL222FCN(X) computes the value of the Schwefel 2.22
% function at point X. SCHWEFEL222FCN accepts a matrix of size M-by-N and
% returns a vetor SCORES of size M-by-1 in which each row contains the
% function value for the corresponding row of X.
% For more information, please visit:
% benchmarkfcns.info/doc/schwefel222fcn
%
% Author: Mazhar Ansari Ardeh
% Please forward any comments or bug reports to mazhar.ansari.ardeh at
% Google's e-mail service or feel free to kindly modify the repository.
function scores = schwefel222fcn(x)

    absx = abs(x);
    scores = sum(absx, 2) + prod(absx, 2);
end
